--- a/ld/configure.tgt
+++ b/ld/configure.tgt
@@ -96,6 +96,10 @@
 aarch64-*-fuchsia*)	targ_emul=aarch64elf
 			targ_extra_emuls="aarch64elfb armelf armelfb"
 			;;
+aarch64-*-ackos*)
+			targ_emul=aarch64ackos
+			targ_extra_emuls="aarch64elf32 aarch64elf32b aarch64elfb armelf armelfb"
+			;;
 aarch64_be-*-linux-gnu_ilp32)
 			targ_emul=aarch64linux32b
 			targ_extra_libpath="aarch64linuxb aarch64linux aarch64linux32 armelfb_linux_eabi armelf_linux_eabi"
@@ -993,6 +997,10 @@
 x86_64-*-haiku*)	targ_emul=elf_x86_64_haiku
 			targ_extra_emuls="elf_x86_64 elf_i386_haiku"
 			;;
+x86_64-*-ackos*)
+			targ_emul=elf_x86_64_ackos
+			targ_extra_emuls="elf_x86_64 elf_i386"
+			;;
 x86_64-*-linux-gnux32)	targ_emul=elf32_x86_64
 			targ_extra_emuls="elf_x86_64 elf_i386 elf_iamcu"
 			targ_extra_libpath="elf_i386 elf_iamcu elf_x86_64"
@@ -1138,6 +1146,11 @@
   NATIVE_LIB_DIRS='/usr/local/lib /usr/ccs/lib /lib /usr/lib'
   ;;
 
+*-*-ackos*)
+  NATIVE_LIB_DIRS='/usr/lib/static /usr/lib/dynamic'
+  ;;
+
+
 esac
 
 case "${target}" in
